src/assetmanagement.rs

```diff
@@ -98,3 +99,73 @@ where
         handle
     }
 }
+
+pub mod util {
+    use amethyst::{
+        assets::{AssetStorage, Loader},
+        ecs::{Entity, ReadExpect, Write, WriteStorage},
+        renderer::{
+            Material, MaterialDefaults, Mesh, MeshHandle, ObjFormat, PngFormat, Texture,
+            TextureMetadata,
+        },
+    };
+    use assetmanagement::AssetManager;
+
+    /// A collection of the required resources necessary to attach `Texture` and `MeshHandle` to an entity
+    pub type AssetStorages<'a> = (
+        ReadExpect<'a, Loader>,
+        Write<'a, AssetManager<Mesh>>,
+        WriteStorage<'a, MeshHandle>,
+        Write<'a, AssetStorage<Mesh>>,
+        Write<'a, AssetManager<Texture>>,
+        WriteStorage<'a, Material>,
+        Write<'a, AssetStorage<Texture>>,
+        ReadExpect<'a, MaterialDefaults>,
+    );
+
+    /// Attaches `Material` and `MeshHandle` of specified asset to the entity
+    pub fn insert_into_asset_storages(
+        entity: Entity,
+        asset_name: &str,
+        storages: &mut AssetStorages,
+    ) {
+        let (
+            ref loader,
+            ref mut mesh_manager,
+            ref mut mesh_handles,
+            ref mut mesh_storage,
+            ref mut tex_manager,
+            ref mut mat_storage,
+            ref mut tex_storage,
+            ref default_mat,
+        ) = storages;
```

Comment:
thats actually an idea :) atm i refactor the whole assetmanager thingy, so I would put this whole situation in `ui_map`, where the refactoring takes place.

Reply:
AFAIHaveTried, there are lifetime problems of structs beeing part of the `'a` systemdata of a system when having a `'b` reference to another (`'b`)resource in the world. The compiler cannot determine correct lifetime bounds for `'b` since it has no boundaries in a system.